File: languages/js/src/entity/availability/publish_response.ts
import * as proto from "../../bridge/proto/bloock_availability";
import { IpnsKey } from "./ipns_key";

/**
 * Represents an object with a the publish response attributes.
 */
export class PublishResponse {
    id: string;
    ipnsKey?: IpnsKey;
  
    /**
     * Constructs a PublishResponse object with the specified parameters.
     * @param id
     */
    constructor(
      id: string,
      ipnsKey?: IpnsKey
    ) {
      this.id = id;
      if (ipnsKey) {
        this.ipnsKey = ipnsKey
      }
    }
  
    static fromProto(res: proto.PublishResponse): PublishResponse {
      if (res.ipnsKey) {
        return new PublishResponse(res.id, IpnsKey.fromProto(res.ipnsKey));
      }
      return new PublishResponse(res.id);
    }
  }

File: languages/go/entity/availability/publish_response.go
package availability

import (
	"github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"
)

// PublishResponse represents an object with a the publish response attributes.
type PublishResponse struct {
	ID      string
	IpnsKey *IpnsKey
}

// NewIpnsKey constructs a PublishResponse object with the specified parameters.
func NewPublishResponse(id string, ipnsKey *IpnsKey) PublishResponse {
	return PublishResponse{
		ID:      id,
		IpnsKey: ipnsKey,
	}
}

func NewPublishResponseFromProto(res *proto.PublishResponse) PublishResponse {
	return NewPublishResponse(res.Id, NewIpnsKeyFromProto(res.IpnsKey))
}
